lib/modules/platform/github/index.ts

```diff
@@ -672,36 +590,16 @@ function matchesState(state: string, desiredState: string): boolean {
 }
 
 export async function getPrList(): Promise<Pr[]> {
-  logger.trace('getPrList()');
-  if (!config.prList) {
-    logger.debug('Retrieving PR list');
-    let prList: GhRestPr[];
-    try {
-      prList = (
-        await githubApi.getJson<GhRestPr[]>(
-          `repos/${
-            config.parentRepo || config.repository
-          }/pulls?per_page=100&state=all`,
-          { paginate: true }
-        )
-      ).body;
-    } catch (err) /* istanbul ignore next */ {
-      logger.debug({ err }, 'getPrList err');
-      throw new ExternalHostError(err, PlatformId.Github);
-    }
-    config.prList = prList
-      .filter(
-        (pr) =>
-          config.forkMode ||
-          config.ignorePrAuthor ||
-          (pr?.user?.login && config?.renovateUsername
-            ? pr.user.login === config.renovateUsername
-            : true)
-      )
-      .map(coerceRestPr);
-    logger.debug(`Retrieved ${config.prList.length} Pull Requests`);
+  if (!config.prCache) {
+    const repo = config.parentRepo ?? config.repository;
+    const username =
+      !config.forkMode && !config.ignorePrAuthor && config.renovateUsername
+        ? config.renovateUsername
+        : null;
+    config.prCache = await getPrCache(githubApi, repo, username);
   }
-  return config.prList;
+
+  return Object.values(config.prCache);
```

Comment:
we should improve this later, currently it doesn't make much sense to have a cache dictionary if we convert it to a list to find a PR.

Reply:
Okay, I'll improve this _today_